### PR TITLE
Add support for glob in file names for tail plugin

### PR DESCRIPTION
**Pull request by srikanthops**
Add a minimal support for glog.
Use glob (3) call and obtain all the file names that are allowed.
from the file names find the one that has the latest modification time and use that file for tail plugin

### CODE REVIEW

File: src/utils_tail.c [hunk 1]
```diff
@@ -36,15 +38,82 @@ struct cu_tail_s
...
+
+/*
+ * Functino: get_last_modified_stat_from_glob
+ * Args: Glob pattern, stat *, char *file_path
+ * Comments: Find all the files returned by glob and find the latest
+ *           and fill stat_buf and file_path
+ * Return : 0 on Success with stat_buf and file_path
+ */
+int
+get_last_modified_stat_from_glob(const char *glob_pattern, struct stat *stat_buf, char *file_path) {
```
Comment (dbrinegar) [inline src/utils_tail.c]: stat_buf has a known size but file_path is open ended, so to make this safe and consistent we should have a file_path_length if we're using it as a return buffer

File: src/utils_tail.c [hunk 2]
```diff
@@ -36,15 +38,82 @@ struct cu_tail_s
...
+    for (i = 0; i < results.gl_pathc; i++) {
+        memset(&ibuf, 0, sizeof(ibuf));
+        ret = stat(results.gl_pathv[i],  &ibuf);
+        if (ret) {
+            continue;
+        }
+
+        if (!rbuf.st_ino)  {
+            memcpy(&rbuf, &ibuf, sizeof(rbuf));
+            strcpy(file_path, results.gl_pathv[i]);
```
Comment (dbrinegar) [inline src/utils_tail.c]: let's use strncpy to avoid buffer overflow

File: src/utils_tail.c [hunk 3]
```diff
@@ -36,15 +38,82 @@ struct cu_tail_s
...
   FILE *fh;
   struct stat stat_buf;
   int status;
+#ifdef OPSCLARITY_GLOB_SUPPORT
+  char file_path[PATH_MAX];
+#endif
 
   memset (&stat_buf, 0, sizeof (stat_buf));
+#ifdef OPSCLARITY_GLOB_SUPPORT
+  status = get_last_modified_stat_from_glob(obj->file, &stat_buf, (char *)&file_path);
```
Comment (dbrinegar) [inline src/utils_tail.c]: pass  sizeof(file_path)  as limit

File: src/utils_tail.c [hunk 4]
```diff
@@ -36,15 +38,82 @@ struct cu_tail_s
...
+        ret = stat(results.gl_pathv[i],  &ibuf);
+        if (ret) {
+            continue;
+        }
+
+        if (!rbuf.st_ino)  {
+            memcpy(&rbuf, &ibuf, sizeof(rbuf));
+            strcpy(file_path, results.gl_pathv[i]);
+        } else if (rbuf.st_mtime < ibuf.st_mtime)  {
+            strcpy(file_path, results.gl_pathv[i]);
```
Comment (dbrinegar) [inline src/utils_tail.c]: this is the same code as above?  can we just collapse the conditionals

File: src/utils_tail.c (hunk 3 above)
Comment (dbrinegar) [inline src/utils_tail.c]: (char *)&   isn't necessary, I hope.  Can it be removed to make it more clear?

File: src/utils_tail.c [hunk 5]
```diff
@@ -59,7 +128,6 @@ static int cu_tail_reopen (cu_tail_t *obj)
     /* Seek to the beginning if file was truncated */
     if (stat_buf.st_size < obj->stat.st_size)
     {
-      INFO ("utils_tail: File `%s' was truncated.", obj->file);
```
Comment (dbrinegar) [inline src/utils_tail.c]: why remove this?  seems useful

File: src/utils_tail.c [hunk 6]
```diff
@@ -28,6 +28,8 @@
 #include "collectd.h"
 #include "common.h"
 #include "utils_tail.h"
+#define OPSCLARITY_GLOB_SUPPORT YES
```
Comment (dbrinegar) [inline src/utils_tail.c]: YES seems an odd way to define a thing.  Nothing else in the code does it this way.  More conventional:

#define OPSCLARITY_GLOB_SUPPORT 1

File: src/utils_tail.c (hunk 1 above)
Comment (dbrinegar) [inline src/utils_tail.c]: can we name the return buffers explicitly so future readers see right away that these are return values?

return_stat_buf
return_file_path

Comment (srikanthops) [inline src/utils_tail.c]: Fixed...

File: src/utils_tail.c (hunk 2 above)
Comment (srikanthops) [inline src/utils_tail.c]: Fixed..

File: src/utils_tail.c [hunk 7]
```diff
@@ -36,15 +37,81 @@ struct cu_tail_s
...
+
+int 
+globerr(const char *path, int eerrno)
+{
+    ERROR("COLLECTD: %s: %s\n", path, strerror(eerrno));
+    return 1;
+}
+
+/*
+ * Functino: get_last_modified_stat_from_glob
```
Comment (dbrinegar) [inline src/utils_tail.c]: typo

**Pull request merged**